requirements.json

```diff
@@ -0,0 +1,85 @@
+{
+  "$schema": "https://raw.githubusercontent.com/DataDog/auto_inject/refs/heads/main/preload_go/cmd/library_requirements_tester/testdata/requirements_schema.json",
+  "version": 1,
+  "native_deps": {
+    "glibc": [{
+      "arch": "arm",
+      "supported": true,
+      "description": "From ubuntu xenial (16.04)",
+      "min": "2.23"
+    },{
+      "arch": "arm64",
+      "supported": true,
+      "description": "From centOS 7",
+      "min": "2.17"
+    },{
+      "arch": "x64",
+      "supported": true,
+      "description": "From centOS 7",
+      "min": "2.17"
+    },{
+      "arch": "x86",
+      "supported": true,
+      "description": "From debian jessie (8)",
+      "min": "2.19"
+    }],
+    "musl": [{
+      "arch": "arm",
+      "supported": true,
+      "description": "From alpine 3.13"
+    },{
+      "arch": "arm64",
+      "supported": true,
+      "description": "From alpine 3.13"
+    },{
+      "arch": "x64",
+      "supported": true,
+      "description": "From alpine 3.13"
+    },{
+      "arch": "x86",
+      "supported": true,
+      "description": "From alpine 3.13"
+    }]
+  },
+  "deny": [
+    {
+      "id": "npm",
+      "description": "Ignore the npm CLI",
+      "os": null,
+      "cmds": [
+        "**/node",
+        "**/nodejs",
+        "**/ts-node",
+        "**/ts-node-*"
+      ],
+      "args": [{ "args":  ["*/npm-cli.js"], "position": 1}],
```

Comment:
nit: on Windows this path separator will be incorrect, but as SSI is not windows-based yet, this is not an issue. I think the only solution with the current design would be to have a duplicate, windows-only rule 🤔 which almost certainly isn't worth it.

Also, is it possible to pass flags to `node` _prior_ to the `npm-cli.js` arg? e.g.

```
/path/to/node --some-flag /path/npm-cli.js
```

If so, then you might not want the `position` argument 🤔 But tbh, the denylist doesn't handle this particularly well right now

Reply:
> nit: on Windows this path separator will be incorrect, but as SSI is not windows-based yet, this is not an issue. I think the only solution with the current design would be to have a duplicate, windows-only rule 🤔 which almost certainly isn't worth it.

I thought about adding Windows but figured as you pointed out that it's not supported anyway right now. For Windows the rules would actually be quite different as even Node itself has a different binary name. I can add support if needed.

> Also, is it possible to pass flags to node prior to the npm-cli.js arg? e.g.

It would be possible when used directly in that way, but when using the `npm` CLI (which should be basically all usage) it should always be first.

> If so, then you might not want the position argument 🤔 But tbh, the denylist doesn't handle this particularly well right now

I only added it because I wasn't sure of the implications of leaving it out, but I can definitely remove it if there aren't any negative side-effects for doing so.

Reply:
> I thought about adding Windows but figured as you pointed out that it's not supported anyway right now. For Windows the rules would actually be quite different as even Node itself has a different binary name. I can add support if needed.

Yeah, sounds good, I don't think there's any need to add this right now

> I only added it because I wasn't sure of the implications of leaving it out, but I can definitely remove it if there aren't any negative side-effects for doing so.

The negative side effects would be that we would block someone running something like:

```
/path/to/node somethingelse.js --some-arg blah/npm-cli.js
```

> It would be possible when used directly in that way, but when using the npm CLI (which should be basically all usage) it should always be first.

Given you state this, I think that leaving it as-is is correct 👍